### PR TITLE
feat: Interactive Day View

### PR DESCRIPTION
**Pull request by werner-scholtz**
# Description

This PR is adding optional functionality that allows users to select and then resize and reschedule events in the calendar DayView.

## Checklist

- [ X] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [ X] I have followed the [Contributor Guide] when preparing my PR.
- [ X] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [ X] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/SimformSolutionsPvtLtd/flutter_calendar_view/blob/master/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: example/lib/widgets/interactive_day_view_widget.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    this.state,
+    this.width,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return DayView<Event>(
+      key: state,
+      width: width,
+      eventTileBuilder: (date, events, boundary, startDuration, endDuration) =>
```
Comment (werner-scholtz) [inline example/lib/widgets/interactive_day_view_widget.dart:22]: Should I keep this [InteractiveDayViewEventTile] a separate widget or should I integrate it into the DayView and add an option to enable Interactive Tiles ?

File: lib/src/calendar_event_data.dart [hunk 2]
```diff
@@ -96,4 +96,156 @@ class CalendarEventData<T extends Object?> {
 
   @override
   int get hashCode => super.hashCode;
+
```
Comment (werner-scholtz) [inline lib/src/calendar_event_data.dart:99]: So this is the best way I could come up with to find the CalendarEventData when the start/end time keeps changing.

File: lib/src/event_controller.dart [hunk 3]
```diff
@@ -181,6 +201,33 @@ class EventController<T extends Object?> extends ChangeNotifier {
     notifyListeners();
   }
```
Comment (werner-scholtz) [inline lib/src/event_controller.dart:203]: I'm not sure if this direction is acceptable if it isn't can we please discuss how to implement this feature correctly.

File: lib/src/calendar_event_data.dart [hunk 4]
```diff
@@ -96,4 +96,156 @@ class CalendarEventData<T extends Object?> {
...
+  /// Checks if the eventData is the same 
+  /// without checking startTime and endTime.
+  ///
+  /// This is the workaround because [CalendarEventData] does not 
+  /// have a unique ID.
+  /// 
+  /// - It can create a bug when 2 events have are exactly the same except 
+  ///   for the startTime and endTime.
+  ///
+  bool compareWithoutTime(Object other) {
```
Comment (werner-scholtz) [inline lib/src/calendar_event_data.dart:109]: So this is the best way I could come up with to find the CalendarEventData when the start/end time keeps changing.

File: lib/src/event_controller.dart [hunk 5]
```diff
@@ -181,6 +201,33 @@ class EventController<T extends Object?> extends ChangeNotifier {
...
 
+  /// Selected eventData in  [DayView], this can be expanded to include other 
+  /// views.
+  /// 
+  /// This is so that the event can be highlighted and modified by the user.
+  CalendarEventData<T>? selectedEventData;
```
Comment (werner-scholtz) [inline lib/src/event_controller.dart:208]: I'm not sure if this direction is acceptable if it isn't can we please discuss how to implement this feature correctly.

**Comment by werner-scholtz** [comment]
Please give it a try I think it is a very nice feature to have, if this is not how you want it implemented I am willing to discuss and change what needs to be changed.

**Comment by werner-scholtz** [comment]
Please give it a try I think it is a very nice feature to have, if this is not how you want it implemented I am willing to discuss and change what needs to be changed.

**Comment by Jei-sKappa** [comment]
Any news?

**Comment by werner-scholtz** [comment]
Hi, so I'm busy creating my own calendar package (https://pub.dev/packages/kalender) and unfortunately I don't have time to revisit this right now.

**Pull request closed without merging**